Update draft-ietf-cats-framework.md (#103)
ietf-wg-cats/draft-ietf-cats-framework: draft-ietf-cats-framework.md

abbrev: CATS Framework
category: info

> docname: draft-ietf-cats-framework-05

Author: need to modify to latest

Reviewer: No, this is set automatically. Worse, changing this will break CI/CD of the repo

Author: ok

area: Routing area
workgroup: cats
keyword:

>  - Computing Computing-Aware Traffic Steering

Author: OK

 :  A traffic engineering approach {{?RFC9522}} that takes into account the dynamic nature of computing resources and network state to optimize service-specific traffic forwarding towards a given service contact instance. Various relevant metrics may be used to enforce such computing-aware traffic steering policies.

Metric:
 : An information that provides suitable input to a selection mechanism to determine a CATS egress node.

Computing metrics:

>   : Computing metrics are the metrics come from the computing side.

Reviewer: ```suggestion
  : Computing metrics are the metrics that are produced and distributed from the computing side.
```

### CATS Service Metric Agent (C-SMA) {#sec-csma}

> The CATS Service Metric Agent (C-SMA) is a functional component that gathers information about service sites and server resources, as well as the status of the different service instances. The C-SMAs may be located adjacent to the service contact instances, co-located with service contact instances, hosted by, or adjacent to, the Egress CATS-Forwarders ({{sec-ocr}}), etc.

Reviewer: ```suggestion
The CATS Service Metric Agent (C-SMA) is a functional component that gathers information about service sites and server resources, as well as the status of the different service instances. A C-SMA may be co-located or located adjacent to a service contact instance, hosted by or adjacent to an Egress CATS-Forwarder ({{sec-ocr}}), etc.
```

          Service CS-ID 1, contact instance CSCI-ID 3 <metrics>
          Service CS-ID 2, <metrics>
~~~

> {: #fig-cats-example-overlay title="An Example of CATS Metric Distribution in a Distributed Model"}

Reviewer: ```suggestion
{: #fig-cats-example-overlay title="An Example of CATS Metric Dissemination in a Distributed Model"}
```

"distribution in a distributed" is heavy

In the example shown in {{fig-cats-example-overlay}}, the client sends a service access via the network through the "CATS-Forwarder 1", which is an Ingress CATS-Forwarder. Note that, a service access may consist of one or more service packets (e.g., Session Initiation Protocol (SIP) {{?RFC3261}}, HTTP {{?RFC9112}}, IPv6 {{?RFC8200}}, SRv6 {{?RFC8754}} or Real-Time Streaming Protocol (RTSP) {{?RFC7826}}) that carry the CS-ID and potential parameters. The Ingress CATS-Forwarder classifies the packets using the information provided by the CATS classifier (C-TC). When a matching classification entry is found for the packets, the Ingress CATS-Forwarder encapsulates and forwards them to the C-PS selected Egress CATS-Forwarder. When these packets reach the Egress CATS-Forwarder, the outer header of the possible overlay encapsulation will be removed and the inner packets will be sent to the relevant service contact instance.

> Note that multi-homed clients may be connected to multiple CATS infrastructures that may be operated by the same or distinct service providers. This version of the framework does not cover multihoming specifics.

## Service Contact Instance Affinity

> Instance affinity means the packets that belong to a flow associated with a service should always be sent to the same service contact instance. Furthermore, packets of a given flow should be forwarded along the same path to avoid mis-ordering and to prevent the introduction of unpredictable latency variations. Specifically, the same Egress CATS-Forwarder may be solicited to forward the packets.

Reviewer: ```suggestion
Instance affinity means that packets that belong to a flow associated with a service should always be sent to the same service contact instance. Furthermore, packets of a given flow should be forwarded along the same path to avoid mis-ordering and to prevent the introduction of unpredictable latency variations. Specifically, the same Egress CATS-Forwarder may be solicited to forward the packets.
```

Author: 'that the' may be better, but we can leave this for RFC editor
